panic with --status-check=true

As I was testing this on https://github.com/dgageot/container-tools/tree/master/hexagons. It started up fine first with  `skaffold dev --port-forward --status-check=true`, but then I got an panic when changed a line in Title.java:

```
Waiting for deployments to stabilize
 - default:deployment/ui is ready. [2/5 deployment(s) still pending]
 - default:deployment/db is ready. [3/5 deployment(s) still pending]
 - default:deployment/game is ready. [4/5 deployment(s) still pending]
 - default:deployment/ref is ready. [1/5 deployment(s) still pending]
panic: assignment to entry in nil map

goroutine 93 [running]:
github.com/GoogleContainerTools/skaffold/pkg/skaffold/event.(*eventHandler).handle(0x655dfe0, 0xc000a2c120)
        pkg/skaffold/event/event.go:439 +0x8ef
created by github.com/GoogleContainerTools/skaffold/pkg/skaffold/event.(*eventHandler).handleResourceStatusCheckEvent
        pkg/skaffold/event/event.go:340 +0xa8
```

--- a/pkg/skaffold/event/event.go
+++ b/pkg/skaffold/event/event.go
@@ -143,11 +143,8 @@ func emptyStateWithArtifacts(builds map[string]string) proto.State {
 		DeployState: &proto.DeployState{
 			Status: NotStarted,
 		},
-		StatusCheckState: &proto.StatusCheckState{
-			Status:    NotStarted,
-			Resources: map[string]string{},
-		},
-		ForwardedPorts: make(map[int32]*proto.PortEvent),
+		StatusCheckState: emptyStatusCheckState(),
+		ForwardedPorts:   make(map[int32]*proto.PortEvent),
 		FileSyncState: &proto.FileSyncState{
 			Status: NotStarted,
 		},
@@ -507,8 +504,15 @@ func ResetStateOnBuild() {
 func ResetStateOnDeploy() {
 	newState := handler.getState()
 	newState.DeployState.Status = NotStarted
-	newState.StatusCheckState.Status = NotStarted
+	newState.StatusCheckState = emptyStatusCheckState()
 	newState.ForwardedPorts = map[int32]*proto.PortEvent{}
 	newState.DebuggingContainers = nil
 	handler.setState(newState)
+}
+
+func emptyStatusCheckState() *proto.StatusCheckState {
+	return &proto.StatusCheckState{
+		Status:    NotStarted,
+		Resources: map[string]string{},
+	}
 }

--- a/pkg/skaffold/event/event_test.go
+++ b/pkg/skaffold/event/event_test.go
@@ -22,6 +22,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/google/go-cmp/cmp/cmpopts"
+
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
 	"github.com/GoogleContainerTools/skaffold/proto"
 	"github.com/GoogleContainerTools/skaffold/testutil"
@@ -355,10 +357,10 @@ func TestResetStateOnBuild(t *testing.T) {
 			},
 		},
 		DeployState:      &proto.DeployState{Status: NotStarted},
-		StatusCheckState: &proto.StatusCheckState{Status: NotStarted},
+		StatusCheckState: &proto.StatusCheckState{Status: NotStarted, Resources: map[string]string{}},
 		FileSyncState:    &proto.FileSyncState{Status: NotStarted},
 	}
-	testutil.CheckDeepEqual(t, expected, handler.getState())
+	testutil.CheckDeepEqual(t, expected, handler.getState(), cmpopts.EquateEmpty())
 }
 
 func TestResetStateOnDeploy(t *testing.T) {
@@ -388,8 +390,18 @@ func TestResetStateOnDeploy(t *testing.T) {
 				"image1": Complete,
 			},
 		},
-		DeployState:      &proto.DeployState{Status: NotStarted},
-		StatusCheckState: &proto.StatusCheckState{Status: NotStarted},
+		DeployState: &proto.DeployState{Status: NotStarted},
+		StatusCheckState: &proto.StatusCheckState{Status: NotStarted,
+			Resources: map[string]string{},
+		},
 	}
-	testutil.CheckDeepEqual(t, expected, handler.getState())
+	testutil.CheckDeepEqual(t, expected, handler.getState(), cmpopts.EquateEmpty())
+}
+
+func TestEmptyStateCheckState(t *testing.T) {
+	actual := emptyStatusCheckState()
+	expected := &proto.StatusCheckState{Status: NotStarted,
+		Resources: map[string]string{},
+	}
+	testutil.CheckDeepEqual(t, expected, actual, cmpopts.EquateEmpty())
 }

